### PR TITLE
Features/pbz/22.0/370134 suppress type

### PR DESCRIPTION
**Pull request by paolobazzi**

### CODE REVIEW

File: org.eclipse.scout.rt.jackson/src/main/java/org/eclipse/scout/rt/jackson/dataobject/ScoutDataObjectModule.java [hunk 1]
```diff
@@ -125,9 +141,4 @@ public void setupModule(SetupContext context) {
   public int hashCode() {
     return NAME.hashCode();
   }
-
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.jackson/src/main/java/org/eclipse/scout/rt/jackson/dataobject/ScoutDataObjectModule.java:128]: Should hashCode be removed as well then? Not sure when this is used.

File: org.eclipse.scout.rt.jackson/src/main/java/org/eclipse/scout/rt/jackson/dataobject/ScoutDataObjectModuleContext.java [hunk 2]
```diff
@@ -94,15 +100,42 @@ public ScoutDataObjectModuleContext withTypeVersionAttributeName(String typeVers
...
 
+  /**
+   * @see #withSuppressTypeAttribute(boolean)
+   */
+  public boolean isSuppressTypeAttribute() {
+    return BooleanUtility.nvl(get(SUPPRESS_TYPE_ATTRIBUTE_KEY, Boolean.class));
+  }
+
+  /**
+   * Flat to suppress writing type attributes (e.g. '_type' attribute in JSON document) when serializing a data object
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.jackson/src/main/java/org/eclipse/scout/rt/jackson/dataobject/ScoutDataObjectModuleContext.java]: Flat -> Flag

Comment (paolobazzi) [reply org.eclipse.scout.rt.jackson/src/main/java/org/eclipse/scout/rt/jackson/dataobject/ScoutDataObjectModuleContext.java]: Done

File: org.eclipse.scout.rt.jackson/src/main/java/org/eclipse/scout/rt/jackson/dataobject/ScoutDataObjectModule.java (hunk 1 above)
Comment (paolobazzi) [reply org.eclipse.scout.rt.jackson/src/main/java/org/eclipse/scout/rt/jackson/dataobject/ScoutDataObjectModule.java:128]: Seems unused as well, done.

**Review by stephan-merkli — APPROVED** [review]

**Pull request merged**